admin/app/components/solidus_admin/ui/table/component.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::UI::Table::Component < SolidusAdmin::BaseComponent
+  def initialize(rows:, model_class:)
+    @rows = rows
+    @model_class = model_class
+    @columns = []
+  end
+
+  def self.build_for(rows, **args, &block)
+    new(rows: rows, **args).tap(&block)
+  end
+
+  def column(name, class_name: nil, &block)
+    name =
+      case name
+      when Symbol then @model_class.human_attribute_name(name)
+      else name
+      end
+
+    @columns << Column.new(name: name, content: block, class_name: class_name)
+  end
+
+  Column = Struct.new(:name, :content, :class_name, keyword_init: true)
```

Comment:
What are the alternatives at defining a class into a component? What about a simple component for the column as well?

Reply:
> I'm sure we can extract it even later if we see that it might be a good idea.

💯 

Right now the contents of the cells are complete responsibility of the code calling the `ui/table` component, while the overall table layout, borders, etc. are a responsibility of the `ui/table` component itself.

We can certainly render components inside cells, not sure if we'll ever need to have an alternative row component, which is by definition coupled with the table itself, especially with features like batch actions and live search.

Reply:
I think the decision here (components vs DSL) is important as it paves the way for a lot of future work and it has effects on the type of product we want to build. Let's talk about it offline! :slightly_smiling_face: :muscle:

Reply:
My feedback is better given as a follow-up PR :slightly_smiling_face: Please, see https://github.com/solidusio/solidus/pull/5176